src/index/VocabularyGenerator.cpp

```diff
@@ -0,0 +1,125 @@
+#include "./VocabularyGenerator.h"
+#include <unordered_set>
+#include <vector>
+#include <queue>
+#include <string>
+#include <utility>
+#include <fstream>
+#include <iostream>
+
+#include "../util/Log.h"
+
+class PairCompare {
+ public:
+  bool operator()(const std::pair<std::string, size_t>& p1, const std::pair<std::string, size_t>& p2) {
+    return p1.first > p2.first;}
+};
+// ___________________________________________________________________
+void mergeVocabulary(const std::string& basename, size_t numFiles) {
+  size_t bufferSize = 200000;
+  std::vector<std::fstream> infiles;
+  std::vector<std::vector<std::string>> vecs(numFiles);
```

Comment:
This deserves a more telling name like `mergeBuf` or something. Maybe also add a comment on how long the inner vectors will get, if I understand correctly each gets up to lines buffered.  Also maybe in the future if we turn that into a compile time constant we may be able to use `std::array` right? Would save an indirection.

Reply:
Some thoughts on this:
1. Currently, nothing gets buffered at all, since this is way more complicated with the binary format which has the IDs in-place. I'm also not sure, if buffering makes sense in this case because we have to write the id of the word anyway and I think the paging of the file system should provide us some buffering or am I wrong in this case?
2. The number of partial files is not a compile-time constant, and if we use a reasonably big buffer this would be to big for a stack-places std::array.

Let me know what you think so I can refactor this function in a more readable way